packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -377,10 +383,20 @@ class ScaffoldMessengerState extends State<ScaffoldMessenger> with TickerProvide
 
   void _updateScaffolds() {
     for (final ScaffoldState scaffold in _scaffolds) {
-      scaffold._updateSnackBar();
+      if (_isRoot(scaffold))
```

Comment:
`_scaffolds` is a type of `LinkedHashSet`, It is order-preserving, so the first one inserted is the root node, right?

Reply:
In addition, after this change, we only need to cache the root node, right? It seems that the others are not used.

Reply:
The tree could change at anytime, so I think the order could potentially be wrong, and if a non-root scaffold becomes a root scaffold on rebuilding, then we wouldn't want to lose it.

Reply:
Have we considered this scenario, inserting a new scope, in the middle of the tree? did we remove the subtree from the old scope?

Reply:
So, I think, if I am understanding right, that would be ok, since when you call ScaffoldMessenger.of(context), you would get the nearest one.